### PR TITLE
[RLlib] Fix PyTorch A3C / A2C loss function using mixed reduced sum / mean

### PR DESCRIPTION
**Pull request by kingsleykuan**
## Why are these changes needed?

Currently, the PyTorch implementation of A3C / A2C fails to converge (tested on BreakoutNoFrameskip-v4). I've isolated the cause to the loss function which reduces the loss across batches differently for different losses.

```
# rllib/agents/a3c/a3c_torch_policy.py

# Entropy Loss uses Reduce Mean
policy.entropy = dist.entropy().mean()

# Policy Loss uses Reduce Sum
policy.pi_err = -train_batch[Postprocessing.ADVANTAGES].dot(
    log_probs.reshape(-1))

# Value Loss uses Reduce Mean
policy.value_err = nn.functional.mse_loss(
    values.reshape(-1), train_batch[Postprocessing.VALUE_TARGETS])
```

This pull request changes all losses to use reduce sum to match the TensorFlow implementation. Below is the average reward when training A2C on BreakoutNoFrameskip-v4 using hyperparameters from the tuned examples.

- Orange: TensorFlow
- Blue: PyTorch (Unmodified)
- Red: PyTorch (Fixed)

a2c-breakout-average-reward

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]
Great catch!

**Comment by ericl** [comment]
I think you need to run scripts/format.sh though: https://travis-ci.com/github/ray-project/ray/jobs/401365605

**Comment by kingsleykuan** [comment]
@ericl Sorry must have missed it, updated!
